### DISCUSSION TITLE
How to Load Extension? Specifically regexp.dll from SQLean

### DISCUSSION THREAD

**Discussion by skinnyshiro** [sqlitebrowser/sqlitebrowser#3407] [Q&A] [answered]
Hello,
I'd really like to get access to the regexp_replace() function from SQLean's regexp extension. I've gone ahead and downloaded the .dll file, placed it in my Extensions folder, and also selected the extension to be loaded for my database (picture below).

![image](https://github.com/sqlitebrowser/sqlitebrowser/assets/74924673/f8f98ca2-c634-41d5-b25b-22092e818ca2)

However, I still cannot get the regexp_replace() function to appear in my queries. Is there a way to check if an extension has been loaded properly? Am I missing a step? 

Apologies if this is covered in the documentation somewhere but I couldn't find any information about it.

**Comment by justinclift** [comment]
A few thoughts spring to mind for me, hopefully something among them is useful. :smile:

* Does the existing regular expression extension not work for this?
* If you disable the existing regular expression extension (as per checkbox at the bottom of that dialog), does it let the SQLean version load?
* It's probably also worth seeing if anything interesting shows up in the SQL Log when you try opening the database.  There might be useful error messages (etc) in there. :smile:

**Reply by skinnyshiro** [reply, marked as answer]
Hi Justin,
So I was able to get it working. My main confusion was that the functions weren't being formatted bold blue like the other SQL functions, so I assumed they hadn't been loaded. I was able to successfully load and test the extension after playing around with it a bit more. 

Regarding whether the existing regexp extension works for this, I need to remove sections of text that match a certain pattern. Although the REGEXP() function can find this pattern in the text, the REPLACE() function requires a substring with no wildcards. So as far as I can tell, I cannot do what I need to without this additional extension. (Sorry if I misunderstood what you were asking).

**Reply by justinclift** [reply]
Nah, you understood fine.  I'm glad you got it working ok. :smile:

**Reply by skinnyshiro** [reply]
Thank you for all your help! I know most of the questions here are for technical issues, so I appreciate the help with user error :)
